src/focus.service.ts

```diff
@@ -509,6 +509,11 @@ export class FocusService {
   private isFocusable(el: HTMLElement) {
     const role = el.getAttribute('role');
     const tabIndex = el.tabIndex;
+    const visible = window.getComputedStyle(el).display !== 'none';
```

Comment:
That's where the exclusion list comes into play, you only really need to run it after you get an element you want to focus on. If it fails there you add the element to a list of exclusions and go back to try again. Since trying to focus on elements isn't a common case, it may average out to better overall performance.